tests/func/experiments/test_checkpoints.py

```diff
@@ -188,3 +192,73 @@ def test_resume_non_head_checkpoint(
     )
     new_head = first(results)
     assert orig_branch != dvc.experiments.get_branch_by_rev(new_head)
+
+
+@pytest.mark.parametrize("use_url", [True, False])
+def test_auto_push_during_iterations(
+    tmp_dir, scm, dvc, checkpoint_stage, git_upstream, local_remote, use_url
+):
+    # set up remote repo
+    remote = git_upstream.url if use_url else git_upstream.remote
+    git_upstream.scm.fetch_refspecs(str(tmp_dir), ["master:master"])
+
+    # without auto push
+    results = dvc.experiments.run(checkpoint_stage.addressing)
+    exp = first(results)
+    ref_info = first(exp_refs_by_rev(scm, exp))
+    assert git_upstream.scm.get_ref(str(ref_info)) is None
+
+    # add auto push
+    os.environ["DVC_EXP_AUTO_PUSH"] = remote
```

Comment:
all of the test cases that reference this env var should probably use `dvc.env.DVC_EXP_AUTO_PUSH` instead of the raw string